artiq/coredevice/ad9912.py

```diff
@@ -24,9 +24,13 @@ class AD9912:
     :param pll_n: DDS PLL multiplier. The DDS sample clock is
         f_ref/clk_div*pll_n where f_ref is the reference frequency and clk_div
         is the reference clock divider (both set in the parent Urukul CPLD
-        instance).
+        instance). For f_ref below 11MHz, internal SYSCLK PLL doubler is used,
+        and pll_n must be halved.
     :param pll_en: PLL enable bit, set to 0 to bypass PLL (default: 1).
         Note that when bypassing the PLL the red front panel LED may remain on.
+
+    .. note:: For lower than default f_ref, onboard loop filter is not optimal
```

Comment:
And now that's something I couldn't get a straightforward answer to after much deliberation with @MorganTL, thus slightly vague wording. 

The loop filter parameters on the 9912 Urukul board were chosen for a multiplier of 10, so 100MHz sysclk. Technically, for any different frequency it should be considered; and technically then it's not really designed for 125MHz we provide to the DDS either, but we consider it okay. From what I learned, using different sysclk than designed changes the phase margin and bandwidth. While phase margin is defined that it must be bigger than 45-50 degrees to keep the PLL stable (and for that extreme example of 10MHz mentioned earlier it is), but and there seems to be no exact guidelines on determining bandwidth, which determines output noise. Whole system performance depends on the quality of the provided clock, and whether adjusting the loop filter is necessary also depends on the use cases.

In such case maybe it would be better to just refer the user to the datasheet of AD9912.